client/src/components/Market/index.jsx

```diff
@@ -0,0 +1,198 @@
+            id: 4,
+            name: "Raftel",
+            currentPrice: 128,
+            rateOfChange: -44,
+            expectedTrend: "Fluctuating",
+            hasPeak: false,
+            freshPrices: true,
+        },
+        {
+            id: 5,
+            name: "Ozu",
+            currentPrice: 222,
+            rateOfChange: 148,
+            expectedTrend: "Large Spike",
+            hasPeak: true,
+            freshPrices: true,
+        },
+        {
+            id: 6,
+            name: "Hyrule",
+            currentPrice: 367,
+            rateOfChange: 148,
+            expectedTrend: "Decrementing",
+            hasPeak: false,
+            freshPrices: false,
+        },
+        {
+            id: 7,
+            name: "Casatopia",
+            currentPrice: 367,
+            rateOfChange: 148,
+            expectedTrend: "Fluctuating",
+            hasPeak: false,
+            freshPrices: false,
+        },
+    ];
+
+    const sortIslandsByPeak = () => {
+        return islands.sort((a, b) =>
+            a.currentPrice <= b.currentPrice ? 1 : -1
+        );
+    };
+
+    const renderIslandInfo = (island) => {
+        return (
+            <div className="flex-center">
+                <Link
+                    className="link dib flex-column"
+                    to={`/island/${island.id}`}
+                >
+                    <i className="pa2 mb2 f-subheadline fas fa-globe-europe secondary-1 bg-primary-1 br3"></i>
+                    <div className="mt1 ml2 primary-1 tc">{island.name}</div>
+                </Link>
+            </div>
+        );
+    };
+
+    const renderTrend = (expectedTrend, hasPeak) => {
+        return (
+            <div className="trend mb3 flex-center f6 primary-2 i tc nowrap br-pill">
+                {(hasPeak && (
+                    <div className="peak-price btn-shadow pa1 br2 flex items-center">
+                        <div
+                            className={`star-${
+                                expectedTrend.includes("Large")
+                                    ? "large"
+                                    : "small"
+                            }`}
+                        ></div>
+                        <div
+                            className={`peak-${
+                                expectedTrend.includes("Large")
+                                    ? "large"
+                                    : "small"
+                            } mh2 secondary-1 small-caps`}
+                        >
+                            {expectedTrend} Peak
+                        </div>
+                    </div>
+                )) ||
+                    expectedTrend}
+            </div>
+        );
+    };
+
+    const renderCurrentPrice = (currentPrice) => {
+        return (
+            <div className="mb3 primary-1 nowrap tc">
+                <i
+                    className={`pa2 bg-primary-1 br-100 fas fa-${
+                        currentPrice > 180
```

Comment:
This logic could be extracted into a pure function that takes the price as an arg and returns the the icon string. I'm leaning towards that for readability, thoughts?

something like:

```ts
getIcon(currentPrice:number): string {
// do logic here to determine which string to return
return "some-icon-string";
}
```

and the classname can be simplified to be `pa2 bg-primary-1 br-100 fas fa-${getIcon(currentPrice)}`

Reply:
Yesssss - that's much better! On it!!!